src/core/qgsmaphittest.h

```diff
@@ -34,6 +38,10 @@ class CORE_EXPORT QgsMapHitTest
     QgsMapSettings mSettings;
     HitTest mHitTest;
 
+    // optional filter expression, not owned
+    /*const*/ QgsExpression* mExpression;
+    // optional polygon, not owned
+    const QgsGeometry* mPolygon;
```

Comment:
Couldn't it just be a QgsGeometry object now? (given that it's implictly shared)...